File: util/golang/client/client_test.go
package client

import (
	"encoding/json"
	"fmt"
	"testing"

	"github.com/alibabacloud-go/tea/tea"
)

type CloudEvent struct {
	Id                  *string                `json:"id,omitempty" xml:"id,omitempty"`
	Source              *string                `json:"source,omitempty" xml:"source,omitempty"`
	Specversion         *string                `json:"specversion,omitempty" xml:"specversion,omitempty"`
	Type                *string                `json:"type,omitempty" xml:"type,omitempty"`
	Datacontenttype     *string                `json:"datacontenttype,omitempty" xml:"datacontenttype,omitempty"`
	Datacontentencoding *string                `json:"datacontentencoding,omitempty" xml:"datacontentencoding,omitempty"`
	Schemaurl           *string                `json:"schemaurl,omitempty" xml:"schemaurl,omitempty"`
	Subject             *string                `json:"subject,omitempty" xml:"subject,omitempty"`
	Time                *string                `json:"time,omitempty" xml:"time,omitempty"`
	Extensions          map[string]interface{} `json:"extensions,omitempty" xml:"extensions,omitempty"`
	Data                []byte                 `json:"data,omitempty" xml:"data,omitempty"`
}

func (s CloudEvent) String() string {
	return tea.Prettify(s)
}

func (s CloudEvent) GoString() string {
	return s.String()
}

func Test_Serialize(t *testing.T) {
	in := []*CloudEvent{
		&CloudEvent{
			Datacontenttype: tea.String("text/plain"),
			Data:            []byte("test"),
		},
		&CloudEvent{
			Datacontenttype: tea.String("text/json"),
			Data:            []byte("test"),
		},
	}
	var tmp []*CloudEvent
	res := Serialize(in)
	fmt.Printf("%s", res)
	byt, _ := json.Marshal(res)
	json.Unmarshal(byt, &tmp)
	fmt.Println(string(tmp[1].Data))
}

File: csharp/tests/ClientTest.cs
using System;
using System.Collections.Generic;
using System.Text;
using AlibabaCloud.SDK.EventBridge;
using AlibabaCloud.SDK.EventBridge.Models;
using Xunit;

namespace tests
{
    public class ClientTest
    {
        [Fact]
        public void TestAPI()
        {
            Config config = new Config
            {
                AccessKeyId = Environment.GetEnvironmentVariable("EventBridge_AccessKey"),
                AccessKeySecret = Environment.GetEnvironmentVariable("EventBridge_AccessSecret"),
                Endpoint = "11.158.193.236:7001"
            };

            var cloudEvent = new CloudEvent
            {
                Datacontenttype = "application/json",
                Data = Encoding.UTF8.GetBytes("test"),
                Id = "a5074581-7e74-4e4c-868f-47e7afdf8445",
                Source = "acs.oss",
                Specversion = "1.0",
                Type = "oss:ObjectCreated:PostObject",
                Time = "2020-08-24T13:54:05.965Z",
                Subject = "acs:oss:cn-hangzhou:1234567:xls-papk/game_apk/123.jpg",
                Extensions = new Dictionary<string, object> { { "aliyuneventbusname", "demo-bus" } }
            };

            var client = new EventBridgeClient(config);
            var resp = client.PutEvents(new List<CloudEvent> { cloudEvent });

            Assert.NotNull(resp);
        }
    }
}
